cpp/src/arrow/adapters/orc/adapter-test.cc

```diff
@@ -0,0 +1,163 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "arrow/adapters/orc/adapter.h"
+#include "arrow/array.h"
+#include "arrow/io/api.h"
+#include "orc/OrcFile.hh"
+
+#include <gtest/gtest.h>
+
+namespace liborc = orc;
+
+namespace arrow {
+
+const int DEFAULT_MEM_STREAM_SIZE = 100 * 1024 * 1024;
+
+class MemoryOutputStream : public liborc::OutputStream {
+ public:
+  explicit MemoryOutputStream(ssize_t capacity) : name("MemoryOutputStream") {
+    data = new char[capacity];
+    length = 0;
+  }
+
+  ~MemoryOutputStream() override { delete[] data; }
+
+  uint64_t getLength() const override { return length; }
+
+  uint64_t getNaturalWriteSize() const override { return naturalWriteSize; }
+
+  void write(const void* buf, size_t size) override {
+    memcpy(data + length, buf, size);
+    length += size;
+  }
+
+  const std::string& getName() const override { return name; }
+
+  const char* getData() const { return data; }
+
+  void close() override {}
+
+  void reset() { length = 0; }
+
+ private:
+  char* data;
```

Comment:
style: members should be named data_ https://google.github.io/styleguide/cppguide.html#Variable_Names